src/publisher.cpp

```diff
@@ -44,7 +44,14 @@ QString Publisher::topic() const { return QString::fromStdString( publisher_->ge
 
 const QString &Publisher::type() const { return type_; }
 
-quint32 Publisher::queueSize() const { return queue_size_; }
+quint32 Publisher::queueSize() const { return qos_.getQoS().get_rmw_qos_profile().depth; }
+
+void Publisher::setQueueSize( quint32 value )
+{
+  qos_.keep_last( value );
+  advertise();
```

Comment:
Not entirely sure I know what I'm doing here.

`Subscription` does this (but calls `subscribe()`), so I tried to mimic this with `advertise()`.

Reply:
It won't the try_advertise checks if it was advertised already and will not readvertise after advertise shut down the publisher.
That's definitely a bug but it wasn't intended to happen.
I'm not entirely sure why publisher has a `try_advertise`, normally, I've used that when it may take multiple tries or the original method may be called from a thread other than the UI thread.
For example, the subscription does it with a timer if no message type is set because I can't block the UI thread while waiting for the topic to come up to infer the message type.
Both are not the case here, though. I've changed that on the master so that a second `advertise` call will  indeed restart the publisher.

Reply:
If we agree `Publisher` cannot be changed in this way, then I'll just remove the setters. No need for `advertise()` to do anything either in that case.

Reply:
I've removed the setters and related slots.

Reply:
sounds good.